pkg/controller/installation/core_controller.go

```diff
@@ -311,6 +311,11 @@ func fillDefaults(instance *operator.Installation) error {
 		}
 	}
 
+	if instance.Spec.NodeMetricsPort != nil && *instance.Spec.NodeMetricsPort == 0 {
```

Comment:
I'm not sure I understand the intention here. IIUC, if the user explicitly writes in `nodeMetricsPort: 0` then we'll assign port `9081` instead. Is that right?

Reply:
Yes... that's my understanding, we use "0" to mean "use the default value".

Reply:
I was imagining this behavior, really since it's a more intuitive / obvious UX.

- If a port is specified, then we use that port. 
- If no port is specified, then we disable metrics.

Is there really a use-case for "Use whatever port the code has arbitrarily chosen as the default"? @ozdanborne WDYT?

Reply:
+1 what casey said

Reply:
Fixed